### ISSUE TITLE
Implement Reserved Slots

### ISSUE THREAD

**Issue by steegness** [steegness/TeamAssist#1]
Implement reserved slot functionality.  This means:

Have a reservedSlots.json file that shows who the reserved users are.

------The following text below this was scrapped in poker's head.  TG generates a file like this, but we should 
------probably make TA be more agnostic.  Therefore, the source file from TG should be turned into something 
------normalized like JSON somehow.
Have a file of IDs.  
- Each line should have one ID on it, at the beginning.
- After the ID, you can put "//" and add comments, like a user's name or such.
- Have the option to use the old STEAM ID (STEAM_0:0:00000) or the id that NS2 uses (00000000)
  When the server is at some predefined capacity, start checking incoming connections against this list.
  If the user is on the list, kick the player on the server who is NOT on the list who has been on the server the longest.  Have a friendly message.

**Comment by steegness** [comment]
Thought:

Have a Config folder, in which we have files we expect the user to edit, like reservedSlots.json (the actual people who get slots) or reservedSlotsConfig.lua (a file where we let the operator decide on things like slots to hold open, kick messages, etc).

**Comment by steegness** [comment]
OK, I started pseudo-coding this.  Before I push a commit, I want to make sure that I'm doing ok by the structure of the project.

I made a new folder under ta called 'config'.  In this folder are currently two files: reservedSlots.json (the file that says who gets a slot) and config.lua.  The latter file's intent is to be where any features are turned on and off, and where variables we expect the server admin to be playing with live, like the message a user gets when they're kicked for a reserved slot holder.

There is also in the ta folder a reservedSlots.lua, and a matching test_reservedSlots.lua in the ta_tests folder.

Good?
